### ISSUE TITLE
storage/shard.go panics on close()

### ISSUE THREAD

**Issue by black-adder** [xichen2020/eventdb#88]
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x44906ff]

goroutine 66 [running]:
github.com/xichen2020/eventdb/storage.(*dbShard).Close(0xc000352780, 0x0, 0x0)
	/Users/won/gocode/src/github.com/xichen2020/eventdb/storage/shard.go:328 +0xbf
github.com/xichen2020/eventdb/storage.(*dbNamespace).closeShards.func1(0x474d780, 0xc000352780)
	/Users/won/gocode/src/github.com/xichen2020/eventdb/storage/namespace.go:263 +0x3a
created by github.com/xichen2020/eventdb/storage.(*dbNamespace).closeShards
	/Users/won/gocode/src/github.com/xichen2020/eventdb/storage/namespace.go:273 +0xf5
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x44906ff]

goroutine 63 [running]:
github.com/xichen2020/eventdb/storage.(*dbShard).Close(0xc000352500, 0x0, 0x0)
	/Users/won/gocode/src/github.com/xichen2020/eventdb/storage/shard.go:328 +0xbf
github.com/xichen2020/eventdb/storage.(*dbNamespace).closeShards.func1(0x474d780, 0xc000352500)
	/Users/won/gocode/src/github.com/xichen2020/eventdb/storage/namespace.go:263 +0x3a
created by github.com/xichen2020/eventdb/storage.(*dbNamespace).closeShards
	/Users/won/gocode/src/github.com/xichen2020/eventdb/storage/namespace.go:273 +0xf5
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x44906ff]

goroutine 61 [running]:
github.com/xichen2020/eventdb/storage.(*dbShard).Close(0xc000352280, 0xc6e25913b0, 0x47440a0)
	/Users/won/gocode/src/github.com/xichen2020/eventdb/storage/shard.go:328 +0xbf
github.com/xichen2020/eventdb/storage.(*dbNamespace).closeShards.func1(0x474d780, 0xc000352280)
	/Users/won/gocode/src/github.com/xichen2020/eventdb/storage/namespace.go:263 +0x3a
created by github.com/xichen2020/eventdb/storage.(*dbNamespace).closeShards
	/Users/won/gocode/src/github.com/xichen2020/eventdb/storage/namespace.go:273 +0xf5
FAIL	github.com/xichen2020/eventdb/integration	16.476s
```

**Comment by xichen2020** [comment]
Do you have a small repro I can use?

**Comment by black-adder** [comment]
@xichen2020 added stack trace

**Comment by xichen2020** [comment]
Great thanks, taking a look.

**Comment by xichen2020** [comment]
Fixed in #91
